### PR TITLE
fix: 🐛 use constructor() instead of UNSAFE_componentWillMount()

### LINKED ISSUE #62: Replace `UNSAFE_componentWillMount` with `constructor`
This could be remedied by replacing `UNSAFE_componentWillMount` with a `constructor` definition, as that is the modern equivalent of using the now-deprecated `componentWillMount`. It will also avoid limiting your supported versions of `react`, as it's both backwards and forwards compatible.

https://github.com/gaearon/react-side-effect/blob/f1e020719f5cc429b96b260cab80ed51b63421a5/src/index.js#L70

### PR DESCRIPTION
**Pull request by streamich**
Closes https://github.com/gaearon/react-side-effect/issues/62

### CODE REVIEW

**Comment by streamich** [comment]
This change should hopefully remove console warnings like below.

image

**Comment by andreyluiz** [comment]
I am very interested on this PR too. @gaearon, can you see it, please?

**Comment by lourd** [comment]
Hey y'all, thanks for the PR!

I'm concerned that this change may create a bug regarding side effect ordering. The tests don't currently cover the behavior, but please refer to [this discussion](https://github.com/gaearon/react-side-effect/issues/40#issuecomment-367186386) for more context.

Could you please add some tests that cover nesting order to demonstrate that changing from componentWillMount to the constructor won't break anything? I know that's asking a lot more, but it would really help.

File: src/index.js [hunk 1]
```diff
@@ -67,7 +67,8 @@ export default function withSideEffect(
         return recordedState;
       }
 
-      UNSAFE_componentWillMount() {
+      constructor(props, context) {
```
Comment (maxmalov) [inline src/index.js:70]: Are you sure this should be in the constructor instead of `componentDidMount`? Ideally, constructors should not have side-effects

**Comment by gaearon** [comment]
Thanks for the PR, but this doesn’t fix anything. It just silences the warning. Side effects in constructor are exactly as bad as in componentWillMount.

**Pull request closed without merging**